### ISSUE TITLE
Snowflake ALTER USER SET option does not works with spaces

### ISSUE THREAD

**Issue by moreaupascal56** [sqlfluff/sqlfluff#6824]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

As said here: https://github.com/sqlfluff/sqlfluff/blob/3cf5754aaa95ad65786cc59d15d8eedf329496e3/src/sqlfluff/dialects/dialect_snowflake.py#L7504-L7519

Snowflake supports SET with spaces but this is not implemented and prefers commas, but does not enforce commas as a rule. Therefore spaces should be linted as well.

Example SQL:

This works:
```sql
ALTER USER "John" SET DEFAULT_ROLE = JOHNNY, DEFAULT_WAREHOUSE = FEU;
```

BUT this do not work: 

```sql
ALTER USER "John" SET DEFAULT_ROLE = JOHNNY DEFAULT_WAREHOUSE = FEU;
```

returns 
```
➜  github sqlfluff lint tmp.sql --dialect snowflake
== [tmp.sql] FAIL
L:   1 | P:  45 |  PRS | Line 1, Position 45: Found unparsable section:
                       | 'DEFAULT_WAREHOUSE = FEU;'
L:   2 | P:   1 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
```

### Expected Behaviour

it should work

### Observed Behaviour

see above

### How to reproduce

see above

### Dialect

snowflake

### Version

3.4.0

### Configuration

none

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
